akka-jackson/src/main/resources/reference.conf

```diff
@@ -0,0 +1,55 @@
+############################
+# Akka Jackson Config File #
+############################
+
+# This is the reference config file that contains all the default settings.
+# Make your edits/overrides in your application.conf.
+
+akka.jackson {
+
+  #//#jackson-modules
+  # The Jackson JSON serializer will register these modules.
+  # It is also possible to use jackson-modules = ["*"] to dynamically
+  # find and register all modules in the classpath.
+  jackson-modules += "com.fasterxml.jackson.module.paramnames.ParameterNamesModule"
+  jackson-modules += "com.fasterxml.jackson.datatype.jdk8.Jdk8Module"
+  jackson-modules += "com.fasterxml.jackson.datatype.jsr310.JavaTimeModule"
```

Comment:
By default, [`JavaTimeModule`](https://fasterxml.github.io/jackson-modules-java8/javadoc/datetime/2.9/com/fasterxml/jackson/datatype/jsr310/JavaTimeModule.html) serializes dates/times as numeric timestamps, which is a kind of weird default for many use cases, and it's something we have thought about changing for Lagom (https://github.com/lagom/lagom/issues/682) to make it use a string rfc3339 format. We haven't yet because it would break compatibility so we would need to come up with a way for users to control this.

On the other hand, maybe a numeric timestamp _is_ better for internal Akka messages. Just raising the issue so it's an explicit decision.

There are a bunch of [`SerializationFeatures`](http://static.javadoc.io/com.fasterxml.jackson.core/jackson-databind/2.9.1/com/fasterxml/jackson/databind/SerializationFeature.html) that control things like this. It's worth considering each, especially because some are hard to change later without breaking things.

Reply:
thx, will consider that

Reply:
It gets even more confusing. While working on [a workaround](https://github.com/ignasi35/painless-microservices/blob/master/reservation-api/src/main/java/com/example/reservation/api/CustomJsonSerializerFactory.java#L20) for [the issue](https://github.com/lagom/lagom/issues/682) I learnt the default formatter for `LocalDate` is an array (which is what the [javadocs](https://fasterxml.github.io/jackson-modules-java8/javadoc/datetime/2.9/com/fasterxml/jackson/datatype/jsr310/JavaTimeModule.html state) but I'm pretty sure we've observed `LocalDate`'s serialised as float representing seconds since epoch. 🤷🏼‍♂️

As @TimMoore mentioned already, a test using golden masters would be great to avoid regressions and clearly document the behavior.